client/state/sites/selectors/has-site-product.ts

```diff
@@ -36,6 +36,6 @@ export default createSelector(
 		if ( ! Array.isArray( productSlug ) ) {
 			productSlug = [ productSlug ];
 		}
-		return `{ siteId || 0 }-${ productSlug.join( '-' ) }`;
+		return `${ siteId }-${ productSlug.join( '-' ) }`;
```

Comment:
I removed the `|| 0` because we do it above already on line 35 (you need to expand the diff to see that).